docs/freqai.md

```diff
@@ -0,0 +1,687 @@
+by a leader with the same `identifier`. In this example, the user has a leader bot with the 
+`identifier: "example"` already running or launching simultaneously as the present follower. 
+The follower will load models created by the leader and inference them to obtain predictions.
+
+### Purging old model data
+
+FreqAI stores new model files each time it retrains. These files become obsolete as new models 
+are trained and FreqAI adapts to the new market conditions. Users planning to leave FreqAI running 
+for extended periods of time with high frequency retraining should set `purge_old_models` in their 
+config:
+
+```json
+    "freqai": {
+        "purge_old_models": true,
+    }
+```
+
+which will automatically purge all models older than the two most recently trained ones.
+
+## Defining model expirations
+
+During dry/live, FreqAI trains each pair sequentially (on separate threads/GPU from the main
+Freqtrade bot). This means there is always an age discrepancy between models. If a user is training
+on 50 pairs, and each pair requires 5 minutes to train, the oldest model will be over 4 hours old. 
+This may be undesirable if the characteristic time scale (read trade duration target) for a strategy 
+is much less than 4 hours. The user can decide to only make trade entries if the model is less than 
+a certain number of hours in age by setting the `expiration_hours` in the config file:
+
+```json
+    "freqai": {
+        "expiration_hours": 0.5,
+    }
+```
+
+In the present example, the user will only allow predictions on models that are less than 1/2 hours
+old. 
+
+## Choosing the calculation of the `target_roi`
+
+As shown in `templates/FreqaiExampleStrategy.py`, the `target_roi` is based on two metrics computed
+by FreqAI: `label_mean` and `label_std`. These are the statistics associated with the labels used 
+*during the most recent training*. This allows the model to know what magnitude of a target to be 
+expecting since it is directly stemming from the training data. By default, FreqAI computes this based 
+on trainig data and it assumes the labels are Gaussian distributed. These are big assumptions 
+that the user should consider when creating their labels. If the user wants to consider the population
+of *historical predictions* for creating the dynamic target instead of the trained labels, the user 
+can do so by setting `fit_live_prediction_candles` to the number of historical prediction candles
+the user wishes to use to generate target statistics. 
+
+```json
+    "freqai": {
+        "fit_live_prediction_candles": 300,
+    }
+```
+
+If the user sets this value, FreqAI will initially use the predictions from the training data set
+and then subsequently begin introducing real prediction data as it is generated. FreqAI will save 
+this historical data to be reloaded if the user stops and restarts with the same `identifier`.
+
+<!-- ## Dynamic target expectation
```

Comment:
?? is this not yet implemented?

Reply:
It is implemented already in `fit_labels()`. The difference is that `fit_labels()` assumes a gaussian distribution in the labels a priori. I have another function which  basically just fits the labels on a bunch of different distribution types and then picks the best fitted one to provide parameters back for. That function is commented out at the end of `data_kitchen.py`.

That approach is just not quite as robust and requires yet more user understanding of an already difficult concept (fitting a statistical distirbution to the trainin labels and using that to understand prediction rarity is a HUGE leap for almost everyone who has tested FreqAI so far. Asking them to now understand quantiles and understand that the return parameters are no longer `mean` and `variance` and are instead based on an esoteric statistical distribution...it is too much for *this PR*. I will reintroduce the idea in a more concise way in the future - and we will start returning `&-s*_stat1` `&-s*_stat2` etc instead of `&-s_mean/std`. 

Further, this auto-distribution fitting approach(commented at the end of `data_kitchen.py` relies on this library "Fitter" which is NOT a library we should/can depend on. I am working on writing our own custom auto-fitter - but that needs better testing and performance considerations.

tl;dr fitting labels with gaussian is good and working well for now. In the future, we will have our own auto-distribution fitter - but not before this merges into develop/stable.